### ISSUE TITLE
source:retrieve doesn't not default to project API version

### ISSUE THREAD

**Issue by ChuckJonas** [forcedotcom/cli#528]
### Summary

In my `sfdx-project.json`, I have version v45.0.  It's also in `manifest/package.xml` (not sure which it makes more sense for it to use; probably actually the manifest).

However, when I run:

`sfdx force:source:retrieve --sourcepath my/project/src/main/default/flows/Create_Deactivation_Case_Flow.flow-meta.xml`

It seems to be using a higher API version (i can tell because it pull properties which are not valid in v45).

I have to explicitly pass the API Version with `-a`, which is obviously not a sustainable workflow as it's impossible to keep all the other developers in sync.

`sfdx force:source:retrieve --sourcepath /Users/charlesjonas/Documents/clients/rs/rs-sfdc/src/main/default/flows/Create_Deactivation_Case_Flow.flow-meta.xml -a 45.0`

If I run `sfdx force:config:list` there is no API version listed

### Steps To Reproduce:

**Repository to reproduce:** N/A. This must be reproduced in a dev or sandbox org.  Any flow should work

***NOTE:** If your issue is not reproducable by dreamhouse-lwc, i.e. requires specific metadata or files, we **require** a link to a simple Salesforce project repository with a script to setup a scratch org that reproduces your problem.*

1.  Create a flow
1.  Setup a project with manifest.  Set API version to 45 in both `sfdx-project.json` and package.xml
1. Retrieve the flow.  Note attributes which are not valid in 45 (eg: `storeOutputAutomatically`)

### Expected result

It should pull the source using the version specified in the `sfdx-project.json` or `manifest/package.xml`

### Actual result

It seems to pull it using a later (guessing latest) version

### Additional information

_Feel free to attach a screenshot_.

**SFDX CLI Version**(to find the version of the CLI engine run sfdx --version):
sfdx-cli/7.66.2-4f159a1d07 darwin-x64 node-v12.13.0

**SFDX plugin Version**(to find the version of the CLI plugin run sfdx plugins --core)

sfdx-cli 7.66.2 (core)

**OS and version**: osx

**Comment by ChuckJonas** [comment]
Also noteworthy.  

- When you pull via the `-x` param, it does use the API version of the manifest.  Seems like for consistency, in projects with manifest, it should grab the API version from there when retrieving any source in the same project

- This also impact the ability to use vscode to retrieve source

**Comment by ChuckJonas** [comment]
Duplicate of https://github.com/forcedotcom/cli/issues/528.

Opened over a year ago... Would be nice if this got some traction...

**Comment by ChuckJonas** [comment]
I linked the wrong issue above (duh) and now I can't seem to find the duplicate I was referencing.  Will re-close if/when found...

This really needs to be fixed as it basically breaks CI every-time salesforce releases a new version of flow

**Comment by dhbahr** [comment]
Indeed .. experiencing it right now, it is infuriating really, that it has made it without a fix this long...

**Comment by YodaDaCoda** [comment]
Just encountered this today when retrieving/deploying a flow. Retrieving the metadata either via force:source:retrieve or the vscode org browser gave me v50 metadata and force:source:deploy tried to deploy with the version set in the project's sourceApiVersion which is v49.

If i specify `--apiversion 49.0` to the retrieve command, I get the expected result, but this should not be necessary.

**Comment by fishygeek91** [comment]
I was able to reproduce this by creating a package.xml and using force:source:retrieve -x “path/to/package.xml” to pull down the a test object in our org. I lowered the API version from 48 to 45 and different data came back with each version. 

Editing the sourceApiVersion variable in sfdx-project.json had no affect. Pulling from an src folder or the Org Browser always returned the same results, no matter then manifest api or the project.json api. 

A fix for this would be very helpful.
